### DISCUSSION TITLE
v4.5.0

### DISCUSSION THREAD

**Discussion by lcharette** [userfrosting/UserFrosting#1151] [Releases]
### Changed Requirements
- Drop PHP 7.1 support. PHP 7.4 is now recommended.
- Raised NodeJS version requirement from `>=10.12.0` to `^12.17.0 || >=14.0.0` (#1138).
- Raised NPM version requirement from `>=6.0.0` to `>=6.14.4` (#1138).

### Changed Composer Dependencies
- Updated `wikimedia/composer-merge-plugin` from `^1.4.0` to `^2.1.0` (#1117).

### Added
- Composer 2 support (#1117).
- [Lando](https://lando.dev) support.
- Added more SMTP options in env and setup:smtp bakery command (#1077),
- Added new `MAIL_MAILER` environment variable to set mailer type.
- Added "Native mail" to `setup:mail` bakery command.

### Changed
- Implement findInt (#1117).
- Replace `getenv()` with `env()` (#1121). 
- Replaced `UserFrosting\Sprinkle\Core\Bakery\Helper\NodeVersionCheck` with new `UserFrosting\Sprinkle\Core\Util\VersionValidator` class.
- Bakery command `setup:smtp` renamed to `setup:mail`. The old command is still available as an alias for backward compatibility. 
- Changed `.php_cs` to `.php_cs.dist`.
- Changed `phpunit.xml` to `phpunit.xml.dist`.

### Fixed
- Replaced AdminLTE credit in default footer (old link was dead).
- Issue with path slashes on Windows (#1133).

### Removed
- Removed deprecated `UserFrosting\System\Bakery\Migration` (deprecated in 4.2.0).
- Removed deprecated `UserFrosting\Tests\DatabaseTransactions` (deprecated in 4.2.0).
- Removed deprecated `UserFrosting\Sprinkle\Core\Tests\ControllerTestCase` (deprecated in 4.2.2).
- Removed deprecated `UserFrosting\Sprinkle\Core\Model\UFModel` (deprecated in 4.1).
- Removed deprecated `UserFrosting\Sprinkle\Core\Sprunje\Sprunje::getResults` (deprecated in 4.1.7).
- Removed deprecated `UserFrosting\Sprinkle\Account\Database\Models\User::exists` (deprecated in 4.1.7).
- Removed deprecated `UserFrosting\Sprinkle\Core\Database\Models\Model::export` (deprecated in 4.1.8).
- Removed deprecated `UserFrosting\Sprinkle\Core\Database\Models\Model::queryBuilder` (deprecated in 4.1.8).
- Removed deprecated `UserFrosting\Sprinkle\Core\Database\Relations\Concerns\Unique::withLimit` (deprecated in 4.1.7).
- Removed deprecated `UserFrosting\Sprinkle\Core\Database\Relations\Concerns\Unique::withOffset` (deprecated in 4.1.7).
- Removed deprecated `UserFrosting\Sprinkle\Core\Error\RendererWhoopsRenderer::getResourcesPath`.
- Removed deprecated `UserFrosting\Sprinkle\Core\Error\RendererWhoopsRenderer::setResourcesPath`.
- Removed deprecated Handlebar `ifCond` (Deprecated in 4.1).
- Removed migration seed.
- Removed support for migration with non static `$dependencies` properties.
- Removed support for deprecated `determineRedirectOnLogin` service (deprecated in 4.1.10).

<hr /><em>This discussion was created from the release <a href='https://github.com/userfrosting/UserFrosting/releases/tag/v4.5.0'>v4.5.0</a>.</em>

**Comment by Silic0nS0ldier** [comment]
Huh. So a discussion can be started in relation to a specific release (that is, with GitHub understanding it relates to a release). Handy.

**Reply by lcharette** [reply]
Yup. New checkbox when you create a release.
